app/src/js/components/Providers/add.js

```diff
@@ -9,23 +9,55 @@ import { isValidProvider } from '../../utils/validate';
 
 const SCHEMA_KEY = 'provider';
 
-const AddProvider = ({ providers }) => (
-  <div className="add_providers">
-    <Helmet>
-      <title> Add Provider </title>
-    </Helmet>
-    <AddRecord
-      baseRoute="/providers/provider"
-      createRecord={createProvider}
-      primaryProperty="id"
-      schemaKey={SCHEMA_KEY}
-      state={providers}
-      title="Create a provider"
-      validate={isValidProvider}
-      validationError={'Concurrent Connection Limit cannot be a negative value'}
-    />
-  </div>
-);
+const nonSftpExcludes = ['privateKey', 'cmKeyId'];
+const nonHttpsExcludes = ['allowedRedirects', 'certificateUri'];
+const s3Excludes = [
+  ...nonSftpExcludes,
+  ...nonHttpsExcludes,
+  'username',
+  'password',
+  'port',
+];
+
+const AddProvider = ({ providers }) => {
+  const [exclude, setExclude] = useState(nonSftpExcludes);
+  const [selectedProtocol, setSelectedProtocol] = useState('http');
+
+  const handleInputChange = {
+    protocol: (value) => {
+      if (value === 's3') setExclude(s3Excludes);
+      else if (value.includes('http')) setExclude(nonSftpExcludes);
+      else if (value === 'sftp') setExclude(nonHttpsExcludes);
+      else setExclude([...nonHttpsExcludes, ...nonSftpExcludes]);
```

Comment:
Is this else statement the same as`value === 'ftp'`. can you make it explicit with no default?  

This did seem a little awkward figuring out what items to exclude rather than what additional items to include with each different protocol.
I would have just...nevermind, looking at how addRecord works, I don't think having includes would be workable.

Reply:
Yeah that's a good call. Made that change. 

And yeah, agreed that the includes would be more intuitive, but then we'd have to include absolutely every field for each one and I was trying to avoid that, especially since fields may get added, etc.